src/torrent.cpp

```diff
@@ -5795,6 +5818,15 @@ namespace {
 			p->update_interest();
 		}
 
+		if (!is_downloading_state(m_state))
+		{
+#ifndef TORRENT_DISABLE_LOGGING
+			debug_log("*** UPDATE_PEER_INTEREST [ skipping, state: %d ]"
+				, int(m_state));
+#endif
+			return;
```

Comment:
update peer interest would also call ``finished()`` or ``resume_download()``, both of which only make sense during downloading, not checking.